x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -293,12 +293,8 @@ export const reindexServiceFactory = (
       // Do any other cleanup work necessary
       reindexOp = await cleanupChanges(reindexOp);
     } else {
-      // Check that it reindexed all documents
-      const {
-        body: { count },
-      } = await esClient.count({ index: reindexOp.attributes.indexName });
-
-      if (taskResponse.task.status!.created < count) {
```

Comment:
This is not a reliable check, as the documents could end up in an `updated` state. This could occur if Kibana is restarted and a user resumes the reindex process in Upgrade Assistant, which actually kicks off the reindex again.

Reply:
More info on the task response body here: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#docs-reindex-api-response-body